cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -0,0 +1,333 @@
+        let responseBody = JSON.stringify({
+            Status: responseStatus,
+            Reason: reasonWithLogInfo,
+            PhysicalResourceId: physicalResourceId || context.logStreamName,
+            StackId: event.StackId,
+            RequestId: event.RequestId,
+            LogicalResourceId: event.LogicalResourceId,
+            Data: responseData,
+        });
+
+        const parsedUrl = new URL(event.ResponseURL);
+        const options = {
+            hostname: parsedUrl.hostname,
+            port: 443,
+            path: parsedUrl.pathname + parsedUrl.search,
+            method: "PUT",
+            headers: {
+                "Content-Type": "",
+                "Content-Length": responseBody.length,
+            },
+        };
+
+        https
+            .request(options)
+            .on("error", reject)
+            .on("response", (res) => {
+                res.resume();
+                if (res.statusCode >= 400) {
+                    reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+                } else {
+                    resolve();
+                }
+            })
+            .end(responseBody, "utf8");
+    });
+}
+
+exports.handler = async function (event, context) {
+    const props = event.ResourceProperties;
+
+    let {LoadBalancerDNS: loadBalancerDNS,
+        LoadBalancerHostedZoneID: loadBalancerHostedZoneID,
+        DomainName: domainName,
+    } = props;
+    const aliases = new Set(props.Aliases);
+
+    acm = new AWS.ACM();
+    envRoute53 = new AWS.Route53();
+    envHostedZoneID = props.EnvHostedZoneId;
+    envName = props.EnvName;
+    appName = props.AppName;
+    serviceName = props.ServiceName;
+    certificateDomain = `${serviceName}-nlb.${envName}.${appName}.${domainName}`;
+
+    // NOTE: If the aliases have changed, then we need to replace the certificate being used, as well as deleting/adding
+    // validation records and A records. In general, any change in aliases indicate a "replacement" of the resources
+    // managed by the custom resource lambda; on the contrary, the same set of aliases indicate that there is no need to
+    // replace or update the certificate, nor the validation records or A records. Hence, we can use this as the physicalResourceID.
+    let aliasesSorted = [...aliases].sort().join(",");
+    const physicalResourceID = `/${serviceName}/${aliasesSorted}`;
+
+    let handler = async function() {
+        switch (event.RequestType) {
+            case "Create":
+                await validateAliases(aliases, loadBalancerDNS);
+                const certificateARN = await requestCertificate({
+                    aliases: aliases,
+                    idempotencyToken: CRYPTO
+                        .createHash("sha256")
+                        .update(physicalResourceID)
+                        .digest("hex")
+                        .substr(0, 32),});
```

Comment:
It probably doesn't really matter since the `idempotencyToken` expires after 1h, but I don't know if it's safe enough to `subtr(0,32)` instead we can do `md5` which gives us exactly 32 digit long hashes and we're not doing anything security related so it should be fine:

```suggestion
                    idempotencyToken: CRYPTO
                        .createHash("md5")
                        .update(physicalResourceID)
                        .digest("hex")});
```